onadata/apps/logger/management/commands/update_attachment_storage_bytes.py

```diff
@@ -0,0 +1,147 @@
+        # to avoid another aggregate query on attachments to calculate users
+        # global usage.
+        total_per_user = defaultdict(int)
+
+        # Release any locks on the users' profile from getting submissions
+        UserProfile.objects.exclude(
+            metadata__attachments_counting_status='complete'
+        ).update(
+            metadata=ReplaceValues(
+                'metadata',
+                updates={'submissions_suspended': False},
+            ),
+        )
+
+        # Get all profiles already updated to exclude their forms from the list
+        up_queryset = UserProfile.objects.values_list('user_id', flat=True).filter(
+            metadata__attachments_counting_status='complete'
+        )
+        # Get only xforms whose users' storage counters have not been updated yet
+        xforms = (
+            XForm.objects.exclude(user_id__in=up_queryset)
+            .values('pk', 'user_id', 'user__username')
+            .order_by('user_id')
+        )
+
+        last_xform = None
+        for xform in xforms:
+
+            if not last_xform or (last_xform['user_id'] != xform['user_id']):
+                # Retrieve or create user's profile.
+                (
+                    user_profile,
+                    created,
+                ) = UserProfile.objects.get_or_create(user_id=xform['user_id'])
+
+                # Some old profiles don't have metadata
+                if user_profile.metadata is None:
+                    user_profile.metadata = {}
+
+                # Set the flag to true if it was never set.
+                if not user_profile.metadata.get('submissions_suspended'):
+                    # We are using the flag `submissions_suspended` to prevent
+                    # new submissions from coming in while the
+                    # `attachment_storage_bytes` is being calculated.
+                    user_profile.metadata['submissions_suspended'] = True
+                    user_profile.save(update_fields=['metadata'])
+
+                # if `last_xform` is not none, it means that the `user_id` is
+                # different from the previous one in the loop so the user
+                # profile must be updated
+                if last_xform:
+                    self.update_user_profile(
+                        last_xform, total_per_user[last_xform['user_id']]
+                    )
+
+            # write out xform progress
+            if self.verbosity >= 1:
+                self.stdout.write(
+                    f"Calculating attachments for xform_id #{xform['pk']}"
+                    f" (user {xform['user__username']})"
+                )
+            # aggregate total media file size for all media per xform
+            form_attachments = Attachment.objects.filter(
+                instance__xform_id=xform['pk']
+            ).aggregate(total=Sum('media_file_size'))
+
+            if form_attachments['total']:
+                with transaction.atomic():
```

Comment:
This code block has moved since the previous review, but I still have the same questions about it: https://github.com/kobotoolbox/kobocat/pull/828#discussion_r953185159

(In short, I think we can ditch `transaction.atomic()` here)

Reply:
Made a mistake, it should be already gone!